### PR TITLE
Wingding Revival

### PR DESCRIPTION
**Pull request by fludd12**
Gives Greys the ability to speak in wingdings again.
:alien: :alien: :alien: :alien: 

TODO:
- [x] Add SQL stuff I think
- [x] Get rid of the spooked comments.
- [x] Pretend to think hard about re-using `speciesprefs` even though it'd hurt my mind body and soul

Also, first PR wooooooo

:cl: Fluff12
rscadd: Greys can now choose to speak in wingdings or not. The Voice option is right below Species in character creation if Grey is selected - make sure you have it set to what you want!
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
👈 👉 👌 🏛 📡

**Comment by fludd12** [comment]
Also, to note: I have no freaking clue how to SQL

At all.

**Comment by FreeStylaLT** [comment]
👌👌👌👌👌

**Comment by Spacemanspark** [comment]
:+1:

**Comment by fludd12** [comment]
Requesting a review of my last commit, if possible - first time doing anything with SQL, sorta just followed what looked like what I was supposed to be doing.

File: SQL/paradise_schema.sql [hunk 1]
```diff
@@ -30,6 +30,7 @@ CREATE TABLE `characters` (
   `age` smallint(4) NOT NULL,
   `species` varchar(45) NOT NULL,
   `language` varchar(45) NOT NULL,
+  `species_span` varchar(20) NOT NULL DEFAULT 'Normal'
```
Comment (tigercat2000) [inline SQL/paradise_schema.sql]: Missing a comma

File: SQL/paradise_schema_prefixed.sql [hunk 2]
```diff
@@ -30,6 +30,7 @@ CREATE TABLE `SS13_characters` (
   `age` smallint(4) NOT NULL,
   `species` varchar(45) NOT NULL,
   `language` varchar(45) NOT NULL,
+  `species_span` varchar(20) NOT NULL DEFAULT 'Normal'
```
Comment (tigercat2000) [inline SQL/paradise_schema_prefixed.sql]: Missing a comma

File: code/modules/client/preference/preferences_mysql.dm [hunk 3]
```diff
@@ -168,6 +168,7 @@
 					socks,
 					body_accessory,
 					gear
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_mysql.dm]: Missing a comma

File: code/modules/client/preference/preferences_mysql.dm [hunk 4]
```diff
@@ -184,6 +185,7 @@
 		age = text2num(query.item[5])
 		species = query.item[6]
 		language = query.item[7]
+		species_span = query.item[66]
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_mysql.dm]: Put this at the end please, these are in order of query, not type

File: code/modules/client/preference/preferences_mysql.dm [hunk 5]
```diff
@@ -458,7 +462,7 @@
 											gen_record,
 											player_alt_titles,
 											disabilities, organ_data, rlimb_data, nanotrasen_relation, speciesprefs,
-											socks, body_accessory, gear)
+											socks, body_accessory, gear,species_span)
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_mysql.dm]: space after comma

File: code/modules/client/preference/preferences_mysql.dm [hunk 6]
```diff
@@ -488,7 +492,7 @@
 											'[sanitizeSQL(gen_record)]',
 											'[playertitlelist]',
 											'[disabilities]', '[organlist]', '[rlimblist]', '[nanotrasen_relation]', '[speciesprefs]',
-											'[socks]', '[body_accessory]', '[gearlist]')
+											'[socks]', '[body_accessory]', '[gearlist]', '[species_span]')
```
Comment (tigercat2000) [inline code/modules/client/preference/preferences_mysql.dm]: this line is completely fine

**Comment by fludd12** [comment]
Better, :tiger: ?

File: code/modules/client/preference/preferences_mysql.dm (hunk 6 above)
Comment (fludd12) [inline code/modules/client/preference/preferences_mysql.dm]: Now I'm scrutinizing the line in case that's sarcasm.

**Comment by fludd12** [comment]
Removing WIP because nobody's found any problems

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 7]
```diff
@@ -123,6 +123,7 @@
 	var/secondary_langs = list()             // The names of secondary languages that are available to this species.
 	var/list/speech_sounds                   // A list of sounds to potentially play when speaking.
 	var/list/speech_chance                   // The likelihood of a speech sound playing.
+	var/list/speech_spans					 // A list of player-optional spans other than normal .say to use when speaking.
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/species/species.dm]: Is this just for admin abuse?

Comment (fludd12) [inline code/modules/mob/living/carbon/human/species/species.dm]: Mostly, yes. I also wanted to make it tracked by-character to make it easier for stuff like clown spans to override it.

**Comment by Krausus** [comment]
You know, instead of touching the SQL to add an entire new preference for the one race with any reason to use a different span class, you *could* just use an existing only-used-for-one-race preference option: `speciesprefs` is currently only used by Vox, but could easily be re-used for a "speak in wingdings" toggle.

**Comment by KasparoVy** [comment]
:+1: Krausus' suggestion, reduce reuse recycle?

**Comment by fludd12** [comment]
I suppose, but I figured this'd be nice for if we decided to do the same thing for other races - IPCs speaking in monospace, for example.

**Comment by fludd12** [comment]
[where doing this man,,,,
where making this happne](http://i.imgur.com/fjeIpVv.png)

**Comment by Spacemanspark** [comment]
I'd love for IPCs to speak in monospace...

**Comment by fludd12** [comment]
This makes that significantly easier, at the very least.

**Comment by Fox-McCloud** [comment]
Reeeee merge commits.

**Comment by fludd12** [comment]
Will get gudder in the future

**Comment by Markolie** [comment]
FYI, the Lavaland PR (which adds the Voice of God) also touches the handle_speech_problems span. In case that's merged before this this implementation may require some changes. Here's the relevant commit: https://github.com/ParadiseSS13/Paradise/pull/6081/commits/46813b285ed702c3050359de5abc82cda831f51f

**Comment by fludd12** [comment]
Oh jeez, I totally forgot that VoG touched that. I'll cross that bridge when we get to it, I guess.

**Comment by fludd12** [comment]
@Markolie If the Lavaland PR gets merged first, I just gotta redo my changes to the file in this PR to ensure there's no conflicts, right...?

**Comment by Markolie** [comment]
I believe so, although it may need a few adjustments. I'll look at it when the time comes!

**Comment by fludd12** [comment]
@Crazylemon64 @tigercat2000 @Markolie Is there anything I should add, or does this look good? Planning on more changes to greys, but they'd be a bit more contentious.

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 8]
```diff
@@ -43,6 +43,7 @@ var/global/default_martial_art = new/datum/martial_art
 	var/icon/lying_icon = null
 
 	var/voice = ""	//Instead of new say code calling GetVoice() over and over and over, we're just going to ask this variable, which gets updated in Life()
+	var/speech_span = "say" // Which span to use for speech. Overridden by mutations.
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/human_defines.dm]: This probably shouldn't be anything by default...

File: code/modules/mob/living/carbon/human/say.dm [hunk 9]
```diff
@@ -143,8 +144,9 @@
 				verb = "yells loudly"
 
 	if((COMIC in mutations) || (locate(/obj/item/organ/internal/cyberimp/brain/clown_voice) in internal_organs))
-		message = "<span class='sans'>[message]</span>"
+		span = "sans"
 
+	message = "<span class='[span]'>[message]</span>"
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/say.dm]: ... and this should be behind an `if(span)` conditional. Saycode already applies a span class, there's no reason to add an extra wrap for humans.

Comment (fludd12) [inline code/modules/mob/living/carbon/human/say.dm]: Fair enough.

**Comment by fludd12** [comment]
Scheiße, this was from the repo I deleted when fixing my own idiotic mistakes. I'll have to fix that.

**Comment by fludd12** [comment]
Opening a new PR because fludd is the biggest idiot ever. This should be the last time I have to do this probably maybe don't count on it

**Pull request closed without merging**